### PR TITLE
Encrypted secrets

### PR DESCRIPTION
**Pull request by zxan1285**
## ✏️ Changes
  Added encryption/decryption features and UI tool for encryption secrets.
  
## 📷 Screenshots
screen shot 2019-02-15 at 16 45 23

## 🔗 References
  Slack: https://auth0.slack.com/archives/C9170558S/p1549925541005200
  
## 🎯 Testing
✅ This change has been tested locally
✅ This change has been tested in a Webtask
✅ This change has unit test coverage
🚫 This change has integration test coverage
🚫 This change has been tested for performance
  
## 🚀 Deployment
  Now, with magic at our side, we are ready to go!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,17 @@ All notable changes to this project will be documented in this file.
...
+## [1.2.0] - Unreleased
+
+### auth0-bitbucket-deploy v2.7.0
+### auth0-github-deploy v2.7.0
+### auth0-gitlab-deploy v2.8.0
+### auth0-visualstudio-deploy v2.6.0
+
+- #### Added
+  - Added support for encoded secrets
+  - Added UI toll for encoding secrets
```
Comment (fyockm) [inline CHANGELOG.md]: s/toll/tool

File: server/lib/slack.js [hunk 2]
```diff
@@ -9,9 +9,11 @@ const createPayload = (report, extensionUrl) => {
     attachments: []
   };
 
+  const providerName = process.env.A0EXT_PROVIDER.charAt(0).toUpperCase() + process.env.A0EXT_PROVIDER.slice(1);
```
Comment (fyockm) [inline server/lib/slack.js]: could `process.env.A0EXT_PROVIDER` be `undefined`? if so, `process.env.A0EXT_PROVIDER.charAt(0)` will throw an error.

Comment (fyockm) [reply server/lib/slack.js]: actually, we _already_ have `lodash` pulled into this file. might as well just use `_.capitalize(process.env.A0EXT_PROVIDER)` and save ourselves the hassle 😉 

https://lodash.com/docs/4.17.11#capitalize

File: server/lib/cipher.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          data[key] = decrypted;
+        }));
+      } else if (typeof data[key] === 'object') {
+        this.prepareData(data[key]);
+      }
+    });
+  }
+
+  processData() {
+    return Promise.all(this.promises);
```
Comment (fyockm) [inline server/lib/cipher.js]: would it make sense for `processData` to accept a `result` param call `prepareData`? 
actually why are they separate? seems like they could be combined as a single method. I'm not a big fan of the private `this.promises` implementation.

File: tests/server/cipher.tests.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import expect from 'expect';
+
+import Cipher from '../../server/lib/cipher';
+
+const password = 'testpwd';
+const cipher = new Cipher(password);
+let encoded;
+
+describe('cipher', () => {
```
Comment (fyockm) [inline tests/server/cipher.tests.js:9]: seems like we're missing unit tests for `decryptString`, `prepareData`, and `processData`

**Review by fyockm — CHANGES_REQUESTED** [review]

File: tests/server/cipher.tests.js (hunk 4 above)
Comment (zxan1285) [reply tests/server/cipher.tests.js:9]: we're doing those as part of `unifyData` tests

**Review by fyockm — APPROVED** [review]

**Pull request merged**